src/main/java/net/imagej/notebook/DefaultNotebookService.java

```diff
@@ -246,6 +260,65 @@ public NotebookTable methods(final Class<?> type, final String prefix) {
 		}
 		return table;
 	}
+	
+	@Override
+	@SuppressWarnings({"unchecked", "rawtypes"})
+	public DefaultDatasetView view(final RandomAccessibleInterval<?> source) {
+
+		final Object element = Util.getTypeFromInterval(source);
+		if (element instanceof ARGBType) {
+			return viewRealType(Converters.argbChannels(
+				(RandomAccessibleInterval<ARGBType>) source, 1, 2, 3));
+		}
+		else if (element instanceof RealType) {
+			return viewRealType((RandomAccessibleInterval<RealType>) source);
+		}
+		else {
+			throw new IllegalArgumentException("Unsupported image type: " + element
+				.getClass().getName());
+		}
+	}
+
+	@Override
+	public DefaultDatasetView view(final RandomAccessibleInterval<?> source,
+		final double min, final double max)
+	{
+		DefaultDatasetView output = view(source);
+		output.setChannelRanges(min, max);
+		output.rebuild();
+		return output;
+	}
+
+	@Override
+	public DefaultDatasetView view(final RandomAccessibleInterval<?> source,
+		final double[] min, final double[] max)
+	{
+
+		if (min.length < source.numDimensions() || max.length < source
+			.numDimensions()) throw new IllegalArgumentException(
+				"Channel maximum and minimum arrays do not match image dimensions!");
+
+		DefaultDatasetView output = view(source);
+
+		for (int c = 0; c < output.getData().dimension(Axes.CHANNEL); c++) {
+			output.setChannelRange(c, min[c], max[c]);
+		}
+
+		output.rebuild();
+		return output;
+	}
+
+	private <T extends RealType<T>> DefaultDatasetView viewRealType(
+		final RandomAccessibleInterval<T> source)
+	{
+		Data data = ds.create(source);
+		DataView view = ids.createDataView(data);
+		if (!(view instanceof DefaultDatasetView))
```

Comment:
Hmm, is there some reason you need to require `DefaultDatasetView` here? That defeats the point of using interfaces.